src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -0,0 +1,599 @@
+  }
+
+  public Future commitOffsets(String group, String instance, final String async, final ConsumerOffsetCommitRequest offsetCommitRequest, final CommitCallback callback) {
+    final KafkaConsumerState state;
+    try {
+      state = getConsumerInstance(group, instance);
+    } catch (RestNotFoundException e) {
+      callback.onCompletion(null, e);
+      return null;
+    }
+
+    return executor.submit(new Runnable() {
+      @Override
+      public void run() {
+        try {
+	    List<TopicPartitionOffset> offsets = state.commitOffsets(async, offsetCommitRequest);
+          callback.onCompletion(offsets, null);
+        } catch (Exception e) {
+          log.error("Failed to commit offsets for consumer " + state.getId().toString(), e);
+          Exception responseException = e;
+          if (!(e instanceof RestException)) {
+            responseException = Errors.kafkaErrorException(e);
+          }
+          callback.onCompletion(null, responseException);
+        } finally {
+          updateExpiration(state);
+        }
+      }
+    });
+  }
+
+  public ConsumerCommittedResponse committed(String group, String instance, ConsumerCommittedRequest request) {
+    log.debug("Committed offsets for consumer " + instance + " in group " + group);
+    ConsumerCommittedResponse response = new ConsumerCommittedResponse();	
+    KafkaConsumerState state = getConsumerInstance(group, instance);
+    if (state != null) {
+	response =  state.committed(request);
+
+    } else {
+
+	response.offsets = new ArrayList<TopicPartitionOffsetMetadata>();	
+    }
+    return response;
+  }
+    
+  public void deleteConsumer(String group, String instance) {
+    log.debug("Destroying consumer " + instance + " in group " + group);
+    final KafkaConsumerState state = getConsumerInstance(group, instance, true);
+    state.close();
+  }
+
+  public void subscribe(String group, String instance, ConsumerSubscriptionRecord subscription) {
+    log.debug("Subscribing consumer " + instance + " in group " + group);
+    KafkaConsumerState state = getConsumerInstance(group, instance);
+    if (state != null) {
+	state.subscribe(subscription);
+    }
+
+  }
+
+  public void unsubscribe(String group, String instance) {
+    log.debug("Unscribing consumer " + instance + " in group " + group);
+    KafkaConsumerState state = getConsumerInstance(group, instance);
+    if (state != null) {
+	state.unsubscribe();
+    }
+
+  }
+
+  public ConsumerSubscriptionResponse subscription(String group, String instance) {
+    log.debug("Unscribing consumer " + instance + " in group " + group);
+    ConsumerSubscriptionResponse response = new ConsumerSubscriptionResponse();
+    KafkaConsumerState state = getConsumerInstance(group, instance);
+    if (state != null) {
+	java.util.Set<java.lang.String> topics =  state.subscription();
+	response.topics = new ArrayList<String>(topics);
+    } else {
+	response.topics = new ArrayList<String>();	
```

Comment:
Yeah, I think this behavior is kind of confusing -- if you get a list of topics back in some cases, not getting it back in others is weird.I'd just drop the return type entirely since it doesn't provide any useful info to the user.